File: problems/CSES/Sorting_and_Searching/playlist/playlist.py
import sys
# sys.setrecursionlimit(200000)

input = sys.stdin.readline

def main():
    n = int(input())
    list_k = list(map(int, input().split()))

    seen = set()
    left = 0
    res = 0

    for right in range(n):
        while list_k[right] in seen:
            seen.remove(list_k[left])
            left += 1

        seen.add(list_k[right])
        res = max(res, right - left + 1)

    print(res)

main()

File: problems/CSES/Sorting_and_Searching/playlist/playlist.cpp
#include <bits/stdc++.h>
using namespace std;
using ll = long long;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int n;
    cin >> n;

    vector<ll> ks(n);
    for (int i = 0; i < n; i++) cin >> ks[i];

    set<ll> seen;
    ll left = 0;
    ll res = 0;

    for (ll right = 0; right < n; right++) {
        while (seen.contains(ks[right])) {
            seen.erase(ks[left]);
            left++;
        }

        seen.insert(ks[right]);
        res = max(res, right - left + 1);
    }

    cout << res << "\n";

    return 0;
}
